modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -2329,9 +2267,6 @@ open class PlaybackManager @Inject constructor(
                 if (isPlaying()) {
                     statsManager.addTotalListeningTime(UPDATE_TIMER_POLL_TIME)
                 }
-                if (playbackStateRelay.blockingFirst().isSleepTimerRunning && !isSleepAfterEpisodeEnabled()) { // Does not apply to end of episode sleep time
-                    setupFadeOutWhenFinishingSleepTimer()
-                }
                 verifySleepTimeForEndOfChapter()
```

Comment:
I could be wrong but would it be better to move the sleep timer logic and state into the SleepTimer class. Changing the code to something like the following:
```suggestion
                val playbackState = playbackStateRelay.blockingFirst()
                if (sleepTimer.shouldPause(playbackState = playbackState)) {
                    pause(sourceView = SourceView.AUTO_PAUSE)
                }
```

Reply:
@geekygecko I like this suggestion and I was trying to push this refactor today, but I am facing with more issues because the `PlaybackManager.setupUpdateTimer` (the place we also verify end of chapter) using RxJava and some methods of this end of chapter code uses Coroutine.

Could we move this refactor to other PR? It would make easier to test and also development. Every refactoring I am doing I need to smoke test the whole sleep timer just because there are a lot of changes here.

Reply:
Sorry I have approved the PR. Feel free to do any refactoring in another PR.

Reply:
No worries. I created this issue for this refactor https://github.com/Automattic/pocket-casts-android/issues/3384